crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -539,7 +539,7 @@ where
 
         // all messages sent in this stream are subprotocol messages, so we need to switch the
         // message id based on the offset
-        compressed[0] = item[0] + this.shared_capability.offset();
+        compressed[0] = item[0] + MAX_RESERVED_MESSAGE_ID + 1;
```

Comment:
is this right? it's not clear to me why we can use a constant here / stop using the offset

Reply:
this is correct because the offset is `reserved + 1`

see how the offsets are determined:

https://github.com/paradigmxyz/reth/blob/e056bfb213e7ddb2f4345d9717735475229024dd/crates/net/eth-wire/src/p2pstream.rs#L630-L633

so what this does here it strips away the reserved offset so that the ID aligns with the first shared capability

Reply:
I'm adding some docs and maybe change the behaviour here so that it perhaps always normalizes messages or also returns the matching capability

Reply:
thx for the docs, yeah, this makes sense. for `eth` plus another capability we'd need to add the offset for the second capability somewhere else right?

Reply:
the offsets are computed after the initial handshake, but atm the caller of the p2p is responsible for matching the message id to a shared cap.
for example, shared caps:: `[("eth", 10), ("other", 5)]` the p2p stream would yield messages with first byte [0..14] (total number of messages), the reserved offset is already subtracted here.
for a single cap the message can be deserialized directly, but with multiple, we'd need to check, for example first byte `12` would actually belong to "other" and we'd need to subtract the number of messages of "eth" before

I'll make this more ergonomic in followups